pkg/sql/catalog/descs/collection.go

```diff
@@ -481,19 +500,89 @@ func (tc *Collection) getDatabaseByName(
 	return true, db, nil
 }
 
+// GetObjectDesc looks up an object by name and returns both its
+// descriptor and that of its parent database. If the object is not
+// found and flags.required is true, an error is returned, otherwise
+// a nil reference is returned.
+//
+// TODO(ajwerner): clarify the purpose of the transaction here. It's used in
+// some cases for some lookups but not in others. For example, if a mutable
+// descriptor is requested, it will be utilized however if an immutable
+// descriptor is requested then it will only be used for its timestamp and to
+// set the deadline.
+func (tc *Collection) GetObjectDesc(
+	ctx context.Context, txn *kv.Txn, db, schema, object string, flags tree.ObjectLookupFlags,
+) (desc catalog.Descriptor, err error) {
+	if isVirtual, desc, err := tc.maybeGetVirtualObjectDesc(
+		schema, object, flags, db,
+	); isVirtual || err != nil {
+		return desc, err
+	}
+	// Resolve type aliases which are usually available in the PostgreSQL as an extension
+	// on the public schema.
+	// TODO(ajwerner): Pull this underneath type resolution.
+	if schema == tree.PublicSchema && flags.DesiredObjectKind == tree.TypeObject {
+		if alias, ok := types.PublicSchemaAliases[object]; ok {
+			if flags.RequireMutable {
+				return nil, errors.Newf("cannot use mutable descriptor of aliased type %s.%s", schema, object)
```

Comment:
Shouldn't this be an `errors.WithAssertionFailedf` instead? If not, shouldn't it have a `pgcode.WrongObjectType`?

Reply:
Done.